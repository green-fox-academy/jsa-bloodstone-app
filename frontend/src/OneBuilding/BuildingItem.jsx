import React from 'react';
import {
  View, Image, Text,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    flexDirection: 'row',
    alignItems: 'center',
    // borderWidth: 1,
    // borderColor: 'red',
  },
  description: {
    marginLeft: 12,
  },
  iconStyle: {
    width: 64,
    height: 64,
  },
  textStyle: {
    fontSize: 14,
    lineHeight: 20,
  },
});

function BuildingItem({
  type, level, getIconImage,
}) {
  return (
    <View style={styles.container}>
      <Image style={styles.iconStyle} resizeMode="contain" source={getIconImage(type)} />
      <View style={styles.description}>
        <Text style={styles.textStyle}>{type}</Text>
        <Text style={styles.textStyle}>{`Level ${level}`}</Text>
      </View>
    </View>
  );
}

BuildingItem.propTypes = {
  type: PropTypes.string.isRequired,
  level: PropTypes.number.isRequired,
  getIconImage: PropTypes.func,
};

BuildingItem.defaultProps = {
  getIconImage: null,
};

export default BuildingItem;
