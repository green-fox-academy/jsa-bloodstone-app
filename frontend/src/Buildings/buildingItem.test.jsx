import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableHighlight } from 'react-native';

import BuildingItem from './buildingItem';

import farmIcon from '../../assets/buildings/factory.png';

const mockGetIconImage = jest.fn(() => farmIcon);

describe('<BuildingItem />', () => {
  afterAll(() => {
    jest.resetModules();
  });

  it('Should render BuildingItem and match the snapshot', () => {
    const mockOnPress = jest.fn(() => { });

    const tree = renderer.create(<BuildingItem
      type="test type"
      level={1}
      onPress={mockOnPress}
      getIconImage={mockGetIconImage}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should call onPress when press BuildingItem', () => {
    const mockOnPress = jest.fn(() => { });

    const testRenderer = renderer.create(<BuildingItem
      type="test type"
      level={1}
      onPress={mockOnPress}
      getIconImage={mockGetIconImage}
    />);
    const testInstance = testRenderer.root;

    testInstance.findByType(TouchableHighlight).props.onPress();
    expect(mockOnPress).toHaveBeenCalled();
  });
});
