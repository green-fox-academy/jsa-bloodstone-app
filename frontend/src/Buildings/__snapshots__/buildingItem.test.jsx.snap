// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildingItem /> Should render BuildingItem and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingVertical": 10,
      "width": 250,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Image
      source={1}
      style={
        Object {
          "height": "auto",
          "maxHeight": 200,
          "maxWidth": 200,
          "minHeight": 70,
          "minWidth": 70,
          "width": "auto",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 15,
        "lineHeight": 20,
      }
    }
  >
    test type
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "lineHeight": 20,
      }
    }
  >
    Level 1
  </Text>
</View>
`;
